lib/runner/VirtualDeviceInvoker.js

```diff
@@ -0,0 +1,81 @@
+const _ = require("lodash");
+const Invoker = require("./Invoker").Invoker;
+const InvokerResponse = require("./Invoker").InvokerResponse;
+const VirtualDevice = require("virtual-device-sdk").VirtualDevice;
+
+module.exports = class VirtualDeviceInvoker extends Invoker {
+    constructor(runner) {
+        super(runner);
+    }
+
+    before(testSuite) {
+        const locale = testSuite.locale || undefined;
+        const voiceId = testSuite.voiceId || undefined;
+        this._virtualDevice = new VirtualDevice(testSuite.virtualDeviceToken, locale, voiceId);
+        this._invocationName = testSuite.invocationName;
+        this._lastTestDescription = "";
+    }
+
+    async invoke(interaction) {
+        if (interaction.test.description !== this._lastTestDescription){
```

Comment:
And thinking about it a bit more, how about adding an index property on the interaction? This can be done in test parser, and then no state is needed on the invoker to track if it is the first interaction